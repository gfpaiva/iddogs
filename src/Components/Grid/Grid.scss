@import '../../Scss/root';

.grid {
  @media screen and (max-width: $sm-beakpoint) { grid-template-columns: repeat(2, 1fr) }
  @media screen and (max-width: $xs-beakpoint) { grid-template-columns: 1fr }
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 1rem;
}
