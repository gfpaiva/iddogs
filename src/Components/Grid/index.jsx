import React from 'react';
import PropTypes from 'prop-types';

import './Grid.scss';
import LazyImage from '../LazyImage';

export default function Grid({ items }) {
  return (
    <div className="grid">
      {items.map((item, idx) => (
        // eslint-disable-next-line react/no-array-index-key
        <LazyImage key={idx} id={idx} image={item} />
      ))}
    </div>
  );
}

Grid.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
};
