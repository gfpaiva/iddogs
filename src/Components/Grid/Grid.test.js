import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import feed from '../../../__mocks__/getFeed';

import Grid from './index';

describe('<Grid />', () => {
  it('should mount properly', () => {
    const wrapper = mount(
      <Router>
        <Grid items={feed.list} />
      </Router>
    );

    expect(wrapper.find(Grid)).toMatchSnapshot();
  })
});
