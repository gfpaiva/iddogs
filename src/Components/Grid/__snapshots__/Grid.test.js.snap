// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> should mount properly 1`] = `
<Grid
  items={
    Array [
      "https://images.dog.ceo/breeds/labrador/n02099712_1150.jpg",
      "https://images.dog.ceo/breeds/labrador/n02099712_1200.jpg",
      "https://images.dog.ceo/breeds/labrador/n02099712_1229.jpg",
      "https://images.dog.ceo/breeds/labrador/n02099712_1254.jpg",
      "https://images.dog.ceo/breeds/labrador/n02099712_129.jpg",
    ]
  }
>
  <div
    className="grid"
  >
    <withRouter(LazyImage)
      id={0}
      image="https://images.dog.ceo/breeds/labrador/n02099712_1150.jpg"
      key="0"
    >
      <LazyImage
        forceVisible={false}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        id={0}
        image="https://images.dog.ceo/breeds/labrador/n02099712_1150.jpg"
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <div
          className="lazy-image"
        >
          <VisibilitySensor
            active={true}
            containment={null}
            delayedCall={false}
            intervalCheck={true}
            intervalDelay={100}
            minTopValue={0}
            offset={
              Object {
                "top": 100,
              }
            }
            onChange={[Function]}
            partialVisibility={true}
            resizeCheck={false}
            resizeDelay={250}
            resizeThrottle={-1}
            scrollCheck={false}
            scrollDelay={250}
            scrollThrottle={-1}
          >
            <CSSTransition
              classNames={
                Object {
                  "enter": "animated fast",
                  "enterActive": "fadeIn fast",
                  "enterDone": "animated fadeIn fast",
                }
              }
              in={false}
              timeout={800}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={800}
                unmountOnExit={false}
              >
                <div
                  className="lazy-image__image"
                  onClick={[Function]}
                  style={Object {}}
                />
              </Transition>
            </CSSTransition>
          </VisibilitySensor>
        </div>
      </LazyImage>
    </withRouter(LazyImage)>
    <withRouter(LazyImage)
      id={1}
      image="https://images.dog.ceo/breeds/labrador/n02099712_1200.jpg"
      key="1"
    >
      <LazyImage
        forceVisible={false}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        id={1}
        image="https://images.dog.ceo/breeds/labrador/n02099712_1200.jpg"
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <div
          className="lazy-image"
        >
          <VisibilitySensor
            active={true}
            containment={null}
            delayedCall={false}
            intervalCheck={true}
            intervalDelay={100}
            minTopValue={0}
            offset={
              Object {
                "top": 100,
              }
            }
            onChange={[Function]}
            partialVisibility={true}
            resizeCheck={false}
            resizeDelay={250}
            resizeThrottle={-1}
            scrollCheck={false}
            scrollDelay={250}
            scrollThrottle={-1}
          >
            <CSSTransition
              classNames={
                Object {
                  "enter": "animated fast",
                  "enterActive": "fadeIn fast",
                  "enterDone": "animated fadeIn fast",
                }
              }
              in={false}
              timeout={800}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={800}
                unmountOnExit={false}
              >
                <div
                  className="lazy-image__image"
                  onClick={[Function]}
                  style={Object {}}
                />
              </Transition>
            </CSSTransition>
          </VisibilitySensor>
        </div>
      </LazyImage>
    </withRouter(LazyImage)>
    <withRouter(LazyImage)
      id={2}
      image="https://images.dog.ceo/breeds/labrador/n02099712_1229.jpg"
      key="2"
    >
      <LazyImage
        forceVisible={false}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        id={2}
        image="https://images.dog.ceo/breeds/labrador/n02099712_1229.jpg"
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <div
          className="lazy-image"
        >
          <VisibilitySensor
            active={true}
            containment={null}
            delayedCall={false}
            intervalCheck={true}
            intervalDelay={100}
            minTopValue={0}
            offset={
              Object {
                "top": 100,
              }
            }
            onChange={[Function]}
            partialVisibility={true}
            resizeCheck={false}
            resizeDelay={250}
            resizeThrottle={-1}
            scrollCheck={false}
            scrollDelay={250}
            scrollThrottle={-1}
          >
            <CSSTransition
              classNames={
                Object {
                  "enter": "animated fast",
                  "enterActive": "fadeIn fast",
                  "enterDone": "animated fadeIn fast",
                }
              }
              in={false}
              timeout={800}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={800}
                unmountOnExit={false}
              >
                <div
                  className="lazy-image__image"
                  onClick={[Function]}
                  style={Object {}}
                />
              </Transition>
            </CSSTransition>
          </VisibilitySensor>
        </div>
      </LazyImage>
    </withRouter(LazyImage)>
    <withRouter(LazyImage)
      id={3}
      image="https://images.dog.ceo/breeds/labrador/n02099712_1254.jpg"
      key="3"
    >
      <LazyImage
        forceVisible={false}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        id={3}
        image="https://images.dog.ceo/breeds/labrador/n02099712_1254.jpg"
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <div
          className="lazy-image"
        >
          <VisibilitySensor
            active={true}
            containment={null}
            delayedCall={false}
            intervalCheck={true}
            intervalDelay={100}
            minTopValue={0}
            offset={
              Object {
                "top": 100,
              }
            }
            onChange={[Function]}
            partialVisibility={true}
            resizeCheck={false}
            resizeDelay={250}
            resizeThrottle={-1}
            scrollCheck={false}
            scrollDelay={250}
            scrollThrottle={-1}
          >
            <CSSTransition
              classNames={
                Object {
                  "enter": "animated fast",
                  "enterActive": "fadeIn fast",
                  "enterDone": "animated fadeIn fast",
                }
              }
              in={false}
              timeout={800}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={800}
                unmountOnExit={false}
              >
                <div
                  className="lazy-image__image"
                  onClick={[Function]}
                  style={Object {}}
                />
              </Transition>
            </CSSTransition>
          </VisibilitySensor>
        </div>
      </LazyImage>
    </withRouter(LazyImage)>
    <withRouter(LazyImage)
      id={4}
      image="https://images.dog.ceo/breeds/labrador/n02099712_129.jpg"
      key="4"
    >
      <LazyImage
        forceVisible={false}
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        id={4}
        image="https://images.dog.ceo/breeds/labrador/n02099712_129.jpg"
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <div
          className="lazy-image"
        >
          <VisibilitySensor
            active={true}
            containment={null}
            delayedCall={false}
            intervalCheck={true}
            intervalDelay={100}
            minTopValue={0}
            offset={
              Object {
                "top": 100,
              }
            }
            onChange={[Function]}
            partialVisibility={true}
            resizeCheck={false}
            resizeDelay={250}
            resizeThrottle={-1}
            scrollCheck={false}
            scrollDelay={250}
            scrollThrottle={-1}
          >
            <CSSTransition
              classNames={
                Object {
                  "enter": "animated fast",
                  "enterActive": "fadeIn fast",
                  "enterDone": "animated fadeIn fast",
                }
              }
              in={false}
              timeout={800}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={800}
                unmountOnExit={false}
              >
                <div
                  className="lazy-image__image"
                  onClick={[Function]}
                  style={Object {}}
                />
              </Transition>
            </CSSTransition>
          </VisibilitySensor>
        </div>
      </LazyImage>
    </withRouter(LazyImage)>
  </div>
</Grid>
`;
